IliasDownloaderUniMA/__init__.py

```diff
@@ -127,6 +173,44 @@ def addCourses(self, *iliasids):
 			self.addCourse(iliasid)
 
 
+	def addAllUserCourses(self, excluded_ids=[]):
+		"""
+		Extracts the users subscribed courses and adds them to the course list.
+
+		:param      excluded_ids:  optional ilias ids to ignore
+		:type       excluded_ids:  list
+
+		:returns:   number of courses added
+		:rtype:     int
+		"""
+
+		num_added = 0
+		courses = self.extractUserCourses()
+
+		for course in courses:
+			iliasid = course["iliasid"]
+
+			if iliasid not in excluded_ids:
+				self.addCourse(iliasid)
+				num_added += 1
+
+		return num_added
```

Comment:
Yeah, never was really happy with this method name either. Couldn't think of a better one myself.
addAllCurrentSemesterCourses sounds pretty good, but I think I'd shorten it to addAllSemesterCourses.
If we add a semester parameter, that points by default to the current one, we could make the method a bit more flexible.
This way the user still has the option to download other semesters if they want to.

Reply:
Good idea, I'd agree on something like

``` python
def addAllSemesterCourses(self, only_current_semester=True, exclude_ids=None):
    # ...
```

Reply:
I don't think limiting it to the current semester is neccessary. 
I'd rather go for a string, by default only the current one, potentially another semester(so specify the semester name) and maybe a wildcard if you want, well, all of them.
Otherwise looks good :)

Reply:
And there's another quirk in Ilias: If English is set as the language in the general settings, the semester string 'HWS 2020' is not for all courses translated to 'WT 2020', the same holds for 'FSS 2020' and 'ST 2020', respectively. Please don't ask me why 😅.

I'd be happy with something like

``` python
def __init__(self):
    # ...
    self.current_semester_pattern = # ... determine the current semester via datetime.now()
    # ...
```

``` python 
def addAllSemesterCourses(self, semester_pattern=None, exclude_ids=None):
    if semester_pattern:
        # ....
    else:
        # use self.current_semester_pattern
```   

where `current_semester_pattern` and `semester_pattern` are regex patterns. I guess then it might be a good idea to also add  1-2 further examples in the readme.md.